text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,656 @@
+multiple messages.
+
+There are three different scenarios for injecting a creation context into a message:
+
+1. A user provides a custom creation context for a message, which doesn't refer
+   to any span described in this document. This provides flexibility to users
+   to model custom scenarios. In this case, no other spans besides the "Publish"
+   span should be created. The "Publish" span should link to the provided
+   creation context.
+2. If no custom creation context is provided for a message, it is recommended
+   to create a "Create" span for every single message and inject its context
+   into the message.  "Create" spans can be created during the "Publish" operation
+   as children of the "Publish" span.  Alternatively, "Create" spans can be
+   created independently of the "Publish" operation, e. g. in cases where messages
+   are created before they are passed to a "Publish" operation. In this case, the
+   "Publish" span should link to the "Create" spans.
+3. For single-message scenarios and when large number of spans are a problem,
+   the context of the "Publish" span can be injected into the message, thus
+   acting as the creation context. In this case, no other spans besides the
+   "Publish" span should be created.
+
+### Consumer
+
+For many use cases, it is not possible to rely on the presence of "Process"
+spans for correlating producer with consumer traces: there are cases where a
+dedicated processing operation cannot be identified, or where processing
+happens in a different trace. Furthermore, processing operations are not often
+covered by messaging libraries and SDKs, but take place in application code.
+Consistently creating spans for "Processing" operations would require either
+effort from the application owner to correctly instrument those operations, or
+additional capabilities of messaging libraries and SDKs (e. g. hooks for
+processing callbacks, which can then be instrumented by the libraries or SDKs).
+
+While it is possible to create "Process" spans and correlate those with
+consumer traces in certain cases, this is not something that can be generally
+required. Therefore, it is more feasible to require the creation of "Deliver"
+spans (for push-based APIs) or "Receive" spans (for pull-based APIs) to
+correlate producer with consumer traces.
+
+#### Instrumenting push-based scenarios
+
+In push-based consumer scenarios, the delivery of messages is not initiated by
+the application code. Instead, callbacks or handlers are registered and then
+called by messaging SDKs to forward messages to the application.
+
+A "Deliver" span covers the call of such a callback or handler and should link
+to the creation context of all messages that are forwarded via the respective
+call.
+
+#### Instrumenting pull-based scenarios
+
+In pull-based consumer scenarios, the delivery of messages is requested by the
+application code. This usually involves a blocking call, which returns zero or
```

Comment:
```suggestion
In pull-based consumer scenarios, the message is actively requested by the
application code. This usually involves a blocking call, which returns zero or
```

To not use the term "delivery" in the non-delivery case.

Could also use "fetched" instead of "requested".